File: 1.1 Modular Programming/example_code_R/modules/exercises/answers/analysis.r


#' Function to groupby and calculate the mean of two columns
aggregate_mean <- function(dataframe, groupby_column, statistic_column) {
  
  # Perform aggregation and summary
  
  # Use group_by_ because of variable column name
  region_mean_density_grouped <- group_by_(.data = dataframe, groupby_column)
  
  # use get() to access column name
  region_mean_density <- dplyr::summarise(.data = region_mean_density_grouped,
                                          "mean_population_density" = mean(get(statistic_column)))
  
  return(region_mean_density)
}


#' Format the dataframe for output
format_frame <- function(dataframe, statistic_column) {
  
  # Sort the data for clearer reading, descending order
  dataframe_sorted <- dplyr::arrange(.data = dataframe)
  
  # Round mean density for clearer reading
  dataframe_sorted[statistic_column] <- round(dataframe_sorted[statistic_column],
                                              digits = 2)
  
  return(dataframe_sorted)
}

File: 4. Reproducible Code/mod_prog_py_r/course_content/example_code_python/modules/exercises/answers/analysis.py
import pandas as pd


def aggregate_mean(dataframe, groupby_column, aggregate_column):
    """Function to groupby and calculate the aggregate mean of two columns"""
    
    dataframe = dataframe.copy()
    
    # Remove unecessary columns
    subset = dataframe[[groupby_column, aggregate_column]]
    
    # Perform mean calculation
    statistic = (subset.groupby(groupby_column, as_index=False)
                       .agg({aggregate_column: "mean"}))
                       
    statistic_renamed = statistic.rename(columns={aggregate_column: "mean_" + aggregate_column})
  
    return statistic_renamed
    
    
    
def format_frame(dataframe, statistic_column):
    """Function to format the dataframe for output"""
    
    dataframe = dataframe.copy()
    
    dataframe_sorted = dataframe.sort_values(by=statistic_column,
                                             ascending=False)
                                      
    dataframe_sorted[statistic_column] = dataframe_sorted[statistic_column].round(2)
    
    return dataframe_sorted
